content/posts/0.27-release.md

```diff
@@ -0,0 +1,251 @@
+---
+title: "Pyodide 0.27 Release"
+date: 2025-01-01
+draft: true
+tags: ["announcement"]
+# author: "Me"
+author: ["Gyeongjae Choi", "Hood Chatham", "Agriya Khetarpal"] # multiple authors
+showToc: true
+TocOpen: false
+draft: false
+hidemeta: false
+comments: false
+# canonicalURL: "https://canonical.url/to/page"
+disableHLJS: true # to disable highlightjs
+disableShare: false
+hideSummary: false
+searchHidden: true
+ShowReadingTime: true
+ShowBreadCrumbs: true
+ShowPostNavLinks: true
+cover:
+    # image: "<image path/url>" # image path/url
+    # alt: "<alt text>" # alt text
+    # caption: "<text>" # display caption under cover
+    relative: false # when using page bundles set this to true
+    hidden: true # only hide on current single page
+---
+
+Pyodide v0.27.0 is out. This release focuses on improving the long-term
+stability of Pyodide.
+
+# Welcome Agriya Khetarpal to the Pyodide team
+
+Agriya Khetarpal has joined Pyodide as the new maintainer. Agriya has been
+active in the Scientific Python area and is a contributor of NumPy, SciPy, and
+scikit-learn. He has already contributed significantly to strengthening Pyodide
+support in various Scientific Python packages.
+
+# Pyodide 0.27 Release
+
+## Build System Improvements
+
+### Decoupling `pyodide-build` from Pyodide runtime
+
+`pyodide-build` is a tool that builds Python packages for the WASM environment to
+run in Pyodide. Previously, the version of `pyodide-build` was strongly tied to
+the version of Pyodide, meaning that if you wanted to build a package against a
+specific version of Pyodide, you had to use a corresponding version of
+`pyodide-build`. The problem with this approach was that even if we improved the
+build system, you would have to wait a few months after Pyodide was released to
+use it.
+
+In this release, we have separated `pyodide-build` from the Pyodide runtime.
+This allows us to develop and release `pyodide-build` independently of Pyodide
+more quickly, and you can apply these improvements to your packages sooner.
+
+pyodide-build is now developed in [pyodide/pyodide-build](https://github.com/pyodide/pyodide-build),
+and you can install it with the command `pip install pyodide-build`.
+
+Recent versions of `pyodide-build` work with Pyodide 0.26 and higher. This means
+you can use, say, `pyodide-build` version 0.29.2, to set up an older Pyodide
+0.26.4 cross-build environment:
+
+```bash
+pip install pyodide-build==0.29.2
+pyodide xbuildenv install 0.26.4
+pyodide build <...>
+```
+
+In case we decide to drop compatibility for a particular version of Pyodide, we
+shall mark it as such in the 
+[cross-build environments metadata file](https://github.com/pyodide/pyodide/blob/main/pyodide-cross-build-environments.json).
```

Comment:
Can we make this sentence more specific? It seems to me that each Pyodide version requires a minimum Pyodide build version, but I don't see a field to specify a maximum Pyodide build version. @agriyakhetarpal do we have a field for that and it's just absent? What will we do if we drop compatibility for a version of Pyodide?

Reply:
Yes, we don't have one for a maximum version yet, because it is perhaps implied that there isn't an "unreleased" Pyodide version that a released version of `pyodide-build` wouldn't support out of the box (we'll catch it in CI). I think we should add one, even if we won't use it actively.

https://github.com/pyodide/pyodide-build/issues/77

Reply:
I think we should delete this sentence because it's not clear to a user what it means. Why do they care that it's marked in the cross-build environments metadata file? Presumably what they care about is that it will be detected if they have a version mismatch. But isn't that assumed?

Reply:
Yes, makes sense to me.